### PR TITLE
Include azure storage library in vendored_sdks

### PR DESCRIPTION
**Pull request by Cyclam**
The aosm extension defined `azure-storage-blob` as a dependency in setup.py's DEPENDENCIES

From CLI team:

> `azure` is a namespace package, adding `azure-xxx` as dependency causes wired issue, for example, not working on MacOS. So the best practice is using vendored sdks.

This PR includes the azure storage library in the aosm extension's vendored_sdk directory.

Using the `azure.multiapi.storage` library was considered, but testing showed inconsistent installation of API versions with the multiapi package, so including the library in vendored_sdks was deemed safer.

## Testing
- Local testing has confirmed that the vendored_sdk version of the modules is being used.
- `azdev tests aosm` passes
- Manual deployment of VNF (with VHD upload using the moved BlobClient library) works

---

This checklist is used to make sure that common guidelines for a pull request are followed.

### Related command
az aosm nfd publish -f input.json --definition-type cnf

### General Guidelines

- [x] Have you run `azdev style <YOUR_EXT>` locally? (`pip install azdev` required)
- [ ] Have you run `python scripts/ci/test_index.py -q` locally? (`pip install wheel==0.30.0` required)

For new extensions:

- [ ] My extension description/summary conforms to the [Extension Summary Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/extensions/extension_summary_guidelines.md).

### About Extension Publish

There is a pipeline to automatically build, upload and publish extension wheels.  
Once your pull request is merged into main branch, a new pull request will be created to update `src/index.json` automatically.  
You only need to update the version information in file setup.py and historical information in file HISTORY.rst in your PR but do not modify `src/index.json`.

### CODE REVIEW

**Comment by yonzhan** [comment]
Thank you for your contribution! We will review the pull request and get back to you soon.

File: src/aosm/HISTORY.rst [hunk 1]
```diff
@@ -8,3 +8,7 @@ Release History
 * Initial release - beta quality
     * `az aosm nfd|nsd generate-config` to generate an example config file to fill in for an NFD or NSD
     * `az aosm nfd|nsd build|publish|delete` to prepare files for, publish or delete an NFD or NSD
+
+1.0.0b2
+++++++++
+* Fixed: Use default_factory when a dataclass default is hashable (Python 3.11 compatibility)
```
[the remark this answers is not in the document]
Comment (Cyclam) [reply src/aosm/HISTORY.rst]: Done

File: src/aosm/HISTORY.rst [hunk 2]
```diff
@@ -11,4 +11,8 @@ Release History
 
 1.0.0b2
 ++++++++
-* Fixed: Use default_factory when a dataclass default is hashable (Python 3.11 compatibility)
\ No newline at end of file
+* Fixed: Use default_factory when a dataclass default is hashable (Python 3.11 compatibility)
+
+1.0.0b3
```
Comment (yanzhudd) [inline src/aosm/HISTORY.rst:16]: It is suggested to list version in descending order in HISTORY.rst. In other words, put the latest version in the top of the file, e.g., https://github.com/Azure/azure-cli-extensions/blob/106a571fb4529a47d97bbe566ca6aa0dcd55b090/src/containerapp/HISTORY.rst

**Comment by yanzhudd** [comment]
please rerun tests in live mode and upload the recording files.

**Comment by bebound** [comment]
LGTM

File: src/aosm/HISTORY.rst (hunk 2 above)
Comment (Cyclam) [reply src/aosm/HISTORY.rst]: Done, thanks.

**Comment by Cyclam** [comment]
> please rerun tests in live mode and upload the recording files.

We no longer have any tests that consume or create any recordings, so I've deleted the recording file.

**Comment by yanzhudd** [comment]
please fix the CI failed item

**Comment by Cyclam** [comment]
> please fix the CI failed item

@yanzhudd Done. Sorry I missed that.

**Comment by yanzhudd** [comment]
please rerun related tests and upload the recording files.

**Review by yanzhudd — APPROVED** [review]

**Comment by Cyclam** [comment]
Hi @yanzhudd , do you know if your team still wants this PR merged?  If not, our team are happy to close without merging, as we prefer to import the storage modules rather than have to maintain them as vendored_sdks anyway :-)

Thanks.

**Comment by Cyclam** [comment]
Hi @yanzhudd , Our team need some certainty on what branch to continue development from.  If this PR is no longer needed, I'm going to close it at the end of the week. If your team does want it merged, could you hit merge before then, please?

Thanks!

**Pull request merged**